### DISCUSSION TITLE
ManyToMany and Indexes

### DISCUSSION THREAD

**Discussion by hayksaryan** [mikro-orm/mikro-orm#4270] [Q&A] [answered]
Suppose tables Author and Book have a many-to-many relationship, and there's an autogenerated AuthorBooks table with 2 columns: `authorId`, `bookId`. 

How do I define an index on both or one of those columns without using a custom pivot table entity?

**Comment by B4nan** [comment, marked as answer]
You could use pivot entity, that way you can define the indexes on pivot table as with a regular entity.

**Reply by hayksaryan** [reply]
Thanks for the quick response. 
So there's no way to accomplish that WITHOUT a custom pivot entity?

**Reply by B4nan** [reply]
Probably not, the autogenerated pivot entity is created here:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/metadata/MetadataDiscovery.ts#L542

The FK properties here:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/metadata/MetadataDiscovery.ts#L616

Keep in mind that you can mix EntitySchema with the other ways to define entity metadata, e.g. you should be able to use it only for the pivot entity, without defining a class for it, just the metadata.

**Reply by hayksaryan** [reply]
Amazing! Thanks for providing the links.

**Reply by B4nan** [reply]
This reminds me I wanted to add some hooks to discovery to allow adjusting the metadata manually.
